server/amp/handlers/story-page.js

```diff
@@ -47,17 +48,24 @@ async function ampStoryPageHandler(
     if (typeof redirectUrls === "function" || (redirectUrls && Object.keys(redirectUrls).length > 0)) {
       await getRedirectUrl(req, res, next, { redirectUrls, config });
     }
+    const story = await Story.getStoryBySlug(client, req.params["0"]);
+    const isAmpDisabled = get(story, ["metadata", "story-attributes", "disable-amp-for-single-story", "0"], "false");
+
+    if ((!isVisualStory && !enableAmp) || isAmpDisabled === "true") {
+      const ampPageBasePath = getAmpPageBasePath(opts, config);
+      const removeString = ampPageBasePath.substring(ampPageBasePath.indexOf("/amp") + "/amp".length + 1);
```

Comment:
For PB clients, the prefix  is customisable and its not just restrained to /amp, we need to add extra code to take that custom value and check with the base url.

Reply:
Shraddha, we are taking the custom amp base path in the above line (55). This `"/amp"` is a fixed value for the amp route. So we have hardcoded it.

Reply:
You are looking for /amp in ampPageBasePath 
As this ampPageBasePath is customisable, users can add even /story-amp for example and this may not work.

Reply:
It cannot have /story-amp as path, the customisable path should start with /amp otherwise it won't match the amp route.

Reply:
PB doesn't restrict the client to add a certain format.
They can add anything in the DB field and it is accepted. So, /story-amp can also be added. 
We need to discuss about this.

Reply:
Sure Shraddha, we can have a discussion around this.